### PR TITLE
fix: clear appointments on course creation

### PR DESCRIPTION
**Pull request by LomyW**
### Description
We have the problem that if the course creation is not completed because, for example, another tab is clicked, the appointments are not cleaned up.

### Solved
Every time we open the `course creation` we clear the `appointments to be created`

### CODE REVIEW

File: src/pages/CreateCourse.tsx [hunk 1]
```diff
@@ -818,6 +819,10 @@ const CreateCourse: React.FC = () => {
         setCurrentIndex(index);
     }, []);
 
+    useEffect(() => {
```
Comment (LomyW) [inline src/pages/CreateCourse.tsx:822]: @timidak that's the only thing I could think of to solve it. Not convinced myself. Do you have another idea?

Comment (timidak) [reply src/pages/CreateCourse.tsx:822]: You could add your code `setAppoiontmensToBeCreated([])` in the cleanup function of the effect above, this should work I guess.

Comment (timidak) [reply src/pages/CreateCourse.tsx:822]: so basically you should add the following to the effect above:
```ts
return () => setAppoiontmensToBeCreated([])
```

File: src/pages/CreateCourse.tsx [hunk 2]
```diff
@@ -496,6 +497,7 @@ const CreateCourse: React.FC = () => {
             // errors === undefined
             if (appointmentsRes.errors) {
                 errors.push('appointments');
+                setAppointmentsToBeCreated([]);
```
Comment (timidak) [inline src/pages/CreateCourse.tsx:500]: By moving this line before the if (but after the `createGroupAppointments`) you could also remove the line  `if (!appointmentsRes.errors) setAppointmentsToBeCreated([]);`.

File: src/pages/CreateCourse.tsx [hunk 3]
```diff
@@ -289,6 +289,7 @@ const CreateCourse: React.FC = () => {
             documentTitle: 'Kurs erstellen',
         });
         // eslint-disable-next-line react-hooks/exhaustive-deps
+        return () => setAppointmentsToBeCreated([]);
```
Comment (timidak) [inline src/pages/CreateCourse.tsx:292]: That worked, right?

**Review by timidak — DISMISSED** [review]
Just a suggestion, but nice work 👍🏻

File: src/pages/CreateCourse.tsx (hunk 3 above)
Comment (LomyW) [reply src/pages/CreateCourse.tsx:292]: yes

**Review by timidak — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**